config/config.go

```diff
@@ -382,11 +382,21 @@ type ScrapeConfig struct {
 	MetricsPath string `yaml:"metrics_path,omitempty"`
 	// The URL scheme with which to fetch metrics from targets.
 	Scheme string `yaml:"scheme,omitempty"`
-	// More than this many samples post metric-relabeling will cause the scrape to fail.
+	// More than this many samples post metric-relabeling will cause the scrape to
+	// fail.
 	SampleLimit uint `yaml:"sample_limit,omitempty"`
 	// More than this many targets after the target relabeling will cause the
 	// scrapes to fail.
 	TargetLimit uint `yaml:"target_limit,omitempty"`
+	// More than this many labels post metric-relabeling will cause the scrape to
+	// fail.
+	LabelLimit uint `yaml:"label_limit,omitempty"`
+	// More than this label name length post metric-relabeling will cause the
+	// scrape to fail.
```

Comment:
Maybe stupid question, but can we fail to scrape the problematic query instead of the whole target? 🤔

Reply:
No we can not. There would be multiple issues with failing only some labels.
It would be inconsistent with all the scrape limits we have.

Also, we do not support half-ingested scrapes and should not. It would lead to incomplete query results at the end for the user.

Reply:
That'd be very difficult for users to deal with correctly, plus would violate the OpenMetrics spec. See https://www.robustperception.io/remote-read-and-partial-failures for the reasoning.

Reply:
Agred, thanks!